backends/memcheck/ceed-memcheck-qfunction.c

```diff
@@ -33,25 +34,33 @@ static int CeedQFunctionApply_Memcheck(CeedQFunction qf, CeedInt Q,
   CeedInt num_in, num_out;
   ierr = CeedQFunctionGetNumArgs(qf, &num_in, &num_out); CeedChkBackend(ierr);
 
-  for (int i = 0; i<num_in; i++) {
+  for (CeedInt i = 0; i<num_in; i++) {
     ierr = CeedVectorGetArrayRead(U[i], CEED_MEM_HOST, &impl->inputs[i]);
     CeedChkBackend(ierr);
   }
-  for (int i = 0; i<num_out; i++) {
+  int mem_block_ids[num_out];
+  for (CeedInt i = 0; i<num_out; i++) {
+    CeedSize len;
+    char name[30] = "";
+
     ierr = CeedVectorGetArrayWrite(V[i], CEED_MEM_HOST, &impl->outputs[i]);
     CeedChkBackend(ierr);
-    CeedSize len;
+
     ierr = CeedVectorGetLength(V[i], &len); CeedChkBackend(ierr);
     VALGRIND_MAKE_MEM_UNDEFINED(impl->outputs[i], len);
+
+    snprintf(name, 30, "'QFunction output %d'", i);
```

Comment:
Just observing: `%d` can only be used with `int`. So long as we're bothering to use `CeedInt` as though it could be a different size, we should decide on the proper way to print it. We could have `CeedInt_FMT` similar to `PetscInt_FMT`. Fortunately, gcc/clang `-Wformat` can check this whenever we try defining `CeedInt` differently.

Reply:
I can add that. I think it makes sense as a follow-up.